### PR TITLE
Regenerate UCUM and FHIRPath lexers/parsers with latest antlr4

### PR DESCRIPTION
**Pull request by lmsurpre**
And update to 4.9.3.

At first I tried to not add the copyright back in as we are neither the authors
of the grammar nor the generated java classes, but that failed our checkstyle
rule and so now I've added those back in.

I also removed the generated tokens and interp files as those are not
needed at runtime (source:
https://stackoverflow.com/questions/59734259/antlr-generated-files).

Commands for regenerating:
```
antlr4 -no-listener -package com.ibm.fhir.model.ucum fhir-model/UCUM.g4
&& sed -i.bak $'s/\t/    /g' fhir-model/UCUM*.java && mv
fhir-model/UCUM*.java fhir-model/src/main/java/com/ibm/fhir/model/ucum/
```
```
antlr4 -visitor -no-listener -package com.ibm.fhir.path fhir-path/FHIRPath.g4 &&
sed -i.bak $'s/\t/    /g' fhir-path/FHIRPath*.java && mv
fhir-path/FHIRPath*.java fhir-path/src/main/java/com/ibm/fhir/path/
```

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public String getDefaultVersion(String url, Class<? extends Resource> resourceTy
      * Get the resource for the given canonical url and resource type
      *
      * @param url
-     *     the canonical url (with optional version postfix)
```
Comment (lmsurpre) [inline fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java:134]: random javadoc update i squeezed in

File: fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathParser.java [hunk 2]
```diff
@@ -152,22 +150,12 @@ public ExpressionContext expression(int i) {
...
     }
     public static class PolarityExpressionContext extends ExpressionContext {
         public ExpressionContext expression() {
             return getRuleContext(ExpressionContext.class,0);
         }
         public PolarityExpressionContext(ExpressionContext ctx) { copyFrom(ctx); }
-        @Override
-        public <T> T accept(ParseTreeVisitor<? extends T> visitor) {
-            if ( visitor instanceof FHIRPathVisitor ) return ((FHIRPathVisitor<? extends T>)visitor).visitPolarityExpression(this);
-            else return visitor.visitChildren(this);
```
Comment (prb112) [inline fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathParser.java:169]: are these now in copyFrom?

**Review by prb112 — APPROVED** [review]
LGTM - discussed offline with Lee

**Comment by prb112** [comment]
> And update to 4.9.3.
> 
> At first I tried to not add the copyright back in as we are neither the authors of the grammar nor the generated java classes, but that failed our checkstyle rule and so now I've added those back in.
> 
> I also removed the generated tokens and interp files as those are not needed at runtime (source: https://stackoverflow.com/questions/59734259/antlr-generated-files).
> 
> Commands for regenerating:
> 
> ```
> antlr4 -no-listener -package com.ibm.fhir.model.ucum fhir-model/UCUM.g4
> && sed -i.bak $'s/\t/    /g' fhir-model/UCUM*.java && mv
> fhir-model/UCUM*.java fhir-model/src/main/java/com/ibm/fhir/model/ucum/
> ```
> 
> ```
> antlr4 -visitor -no-listener -package com.ibm.fhir.path fhir-path/FHIRPath.g4 &&
> sed -i.bak $'s/\t/    /g' fhir-path/FHIRPath*.java && mv
> fhir-path/FHIRPath*.java fhir-path/src/main/java/com/ibm/fhir/path/
> ```
> 
> Signed-off-by: Lee Surprenant [lmsurpre@us.ibm.com](mailto:lmsurpre@us.ibm.com)

https://github.com/IBM/FHIR/wiki/ANTLR-Usage <-- Lee created this.

**Review by tbieste — APPROVED** [review]
Adding the documentation to the Wiki was a nice touch.
LGTM

**Pull request merged**